components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -15,71 +15,304 @@ export class BitbucketServerApi {
     @inject(AuthProviderParams) protected readonly config: AuthProviderParams;
     @inject(BitbucketServerTokenHelper) protected readonly tokenHelper: BitbucketServerTokenHelper;
 
-    public async runQuery<T>(user: User, urlPath: string): Promise<T> {
+    public async runQuery<T>(
+        userOrToken: User | string,
+        urlPath: string,
+        method: string = "GET",
+        body?: string,
+    ): Promise<T> {
+        const token =
+            typeof userOrToken === "string"
+                ? userOrToken
+                : (await this.tokenHelper.getTokenWithScopes(userOrToken, [])).value;
+        const fullUrl = `${this.baseUrl}${urlPath}`;
+        let result: string = "OK";
+        try {
+            const response = await fetch(fullUrl, {
+                timeout: 10000,
+                method,
+                headers: {
+                    "Content-Type": "application/json",
+                    Authorization: `Bearer ${token}`,
+                },
+                body,
+            });
+
+            if (!response.ok) {
+                try {
+                    const result = await response.json();
+                    if (BitbucketServer.ErrorResponse.is(result)) {
+                        throw new Error(`${response.status} / ${result.errors[0]?.message}`);
+                    }
+                } catch {}
+                throw Error(`${response.status} / ${response.statusText}`);
+            }
+            return (await response.json()) as T;
+        } catch (error) {
+            result = "error " + error?.message;
+            throw error;
+        } finally {
+            console.debug(`BBS GET ${fullUrl} - ${result}`);
+        }
+    }
+
+    public async fetchContent(user: User, urlPath: string): Promise<string> {
         const token = (await this.tokenHelper.getTokenWithScopes(user, [])).value;
         const fullUrl = `${this.baseUrl}${urlPath}`;
-        const response = await fetch(fullUrl, {
-            timeout: 10000,
-            method: "GET",
-            headers: {
-                "Content-Type": "application/json",
-                Authorization: `Bearer ${token}`,
-            },
-        });
-        if (!response.ok) {
-            throw Error(response.statusText);
+        let result: string = "OK";
+        try {
+            const response = await fetch(fullUrl, {
+                timeout: 10000,
+                method: "GET",
+                headers: {
+                    Authorization: `Bearer ${token}`,
+                },
+            });
+
+            if (!response.ok) {
+                throw Error(`${response.status} / ${response.statusText}`);
+            }
+            return await response.text();
+        } catch (error) {
+            result = "error " + error?.message;
+            throw error;
+        } finally {
+            console.debug(`BBS GET ${fullUrl} - ${result}`);
```

Comment:
```suggestion
            console.debug(`BitBucketServer GET ${fullUrl} - ${result}`);
```
Just to reduce mental overhead model.